virttest/qemu_vm.py

```diff
@@ -491,19 +491,17 @@ def add_virtio_port(devices, chardev, params, name, bus, index=None):
             :param bus: Which virtio-serial-pci device use
             :param index: Index of the current virtio_port
             """
-            def get_extra_options(params):
-                """Get extra params pairs"""
-                options = dict()
+            def set_extra_options(virtio_port, params):
+                """Set extra params pairs"""
                 extra_params = params.get('virtio_port_params', '')
                 for _ in extra_params.split():
                     try:
                         if "=" not in _:
                             key, value = _, "NO_EQUAL_STRING"
                         else:
                             key, value = _.split('=')
-                        options[key] = value
+                        virtio_port.set_param(key, value)
                     except Exception:
-                        options.clear()
                         msg = ("Invaild params %s in " % _ +
                                "'virtio_port_param' = %s" % extra_params)
                         logging.error(msg)
```

Comment:
thanks your comments and commit updated, would like to review again.

thanks,
Xu